docs/users/step-debugging.md

```diff
@@ -1,90 +1,90 @@
 
-![PhpStorm debug start](images/phpstorm_config_debug_button.png)
+### Debug a command line application
 
-Server creation:
+To start a step debug session for a PHP script on the command line prefix your command line
+with `php -dxdebug.start_with_request=yes {path_to_your_script}` when executing the script inside the ddev container.
 
-![PhpStorm server creation](images/phpstorm_config_server_config.png)
+Alternatively you can run `ddev xdebug autostart` and just execute the script as usual.
 
-#### PhpStorm and Command-Line Debugging
+Your IDE may offer you additional ways to debug for example unit tests.
 
-If you need to debug command-line PHP processes, especially code that is outside the docroot, the environment variable PHP_IDE_CONFIG is already set inside the web container, so you don't have to do much more.
+Please note: When using PHPStorm you should first [debug a website](#Debug a website) and configure a path mapping.
 
-However, if you have not yet used PhpStorm with xdebug for a regular web request, do that to automatically create the PhpStorm "server" with the same name as your primary URL (see "Languages and Frameworks" -> "PHP" -> "Servers"). The key job of the "server" is to map filesystem locations on the workstation (your computer) to filesystem locations on the remote server (in this case the ddev-webserver container). Often, PhpStorm has automatically set up a mapping that doesn't include the entire project (so the vendor directory is not mapped, for example). So map the top-level directory of your project to /var/www/html in the container, as in this image:
-![PhpStorm mapping](images/PHPStormServerMapping.png)
+### Configuration
 
-### Visual Studio Code (vscode) Debugging Setup
+**Key facts:**
 
-1. Install the [php-debug](https://marketplace.visualstudio.com/items?itemName=felixfbecker.php-debug) extension.
-2. Update the project's [launch.json] (in `.vscode/launch.json`)  to add "Listen for xdebug" (see [config snippet](snippets/launch.json)).  For more on launch.json, see [vscode docs](https://code.visualstudio.com/docs/editor/debugging#_launch-configurations).
-3. Set a breakpoint in your index.php. If it isn't solid red, restart.
-4. In the menu, choose Run->Start Debugging.You may have to select "Listen for XDebug" by the green arrowhead at the top left. The bottom pane of vscode should now be orange (live) and should say "Listen for XDebug".
-5. Enable XDebug with `ddev xdebug on`
-6. In a browser, visit your project, you should hit the breakpoint.
+* You can let xdebug start a debugging session for every request with `ddev xdebug autostart` and
+  return to the default with `ddev xdebug trigger`
+* You can disable xdebug with `ddev xdebug off` or setting `xdebug_enabled: false` in your `.ddev/config.yml`
+* `ddev xdebug status` will show you the current xdebug configuration
+* If xdebug is disabled in your configuration, you can enable it for a already-running project with `ddev xdebug on`
```

Comment:
`ddev xdebug autostart` and `ddev xdebug trigger` is explained in the first line. And no, I would not suggest using using xdebug autostart, as it will cause problems if your website executes AJAX Requests, or contains images served by php or CLI scripts run in parallel while you are debugging.
`ddev xdebug autostart` is just a "compatbility" mode for the current `xdebug on` behavior and might be useful for people who do not use an IDE for development.